### PR TITLE
Fix: SDL_PhysFS_LoadFile handles empty files correctly

### LINKED ISSUE #10: Bug: SDL_PhysFS_LoadFile fails silently on empty files
In `SDL_PhysFS.h` line 581, `if (size <= 0)` treats a file of size 0 as a failure and returns NULL with `*datasize=0` — but no SDL error is set, so callers can't tell whether the file was missing or simply empty. `PHYSFS_fileLength` returns -1 on error and 0 for a legitimately empty file.

## Suggested Implementation

Split the check:

```c
if (size < 0) {
    SDL_PhysFS_SetError("Cannot determine file size");
    PHYSFS_close(handle);
    return NULL;
}
// size == 0: allocate a 1-byte null-terminated buffer and return it
```

## QA

Add a test that loads an empty file and verifies a non-NULL pointer is returned, `*datasize == 0`, and no SDL error is set.

### PR DESCRIPTION
**Pull request by RobLoach**
Splits the `size <= 0` guard into two cases:
- `size < 0`: `PHYSFS_fileLength` failed — set an SDL error and return NULL.
- `size == 0`: legitimately empty file — return a non-NULL 1-byte null-terminated buffer with `*datasize == 0` and no SDL error set.

This lets callers distinguish a missing/unreadable file from an empty one using the standard SDL error pattern.

Fixes #10

### CODE REVIEW

File: include/SDL_PhysFS.h [hunk 1]
```diff
@@ -578,14 +578,27 @@ void* SDL_PhysFS_LoadFile(const char* filename, size_t *datasize) {
...
+        SDL_PhysFS_SetError("Cannot determine file size");
         if (datasize != NULL) {
             *datasize = 0;
         }
         PHYSFS_close(handle);
         return NULL;
     }
 
+    // Empty file: return a 1-byte null-terminated buffer so callers get a
+    // non-NULL pointer and can distinguish "empty" from "error".
```
Comment (RobLoach) [inline include/SDL_PhysFS.h]:
``` suggestion
    // Existing empty files get an empty null-terminated buffer.
```
